File: server/router/miser/miser_stat.go
package miser

import "github.com/gin-gonic/gin"

type MiserStatRouter struct {
}

func (s *MiserCategoryRouter) InitMiserStatRouter(privateRouter *gin.RouterGroup, publicRouter *gin.RouterGroup) {
	withoutRecord := privateRouter.Group("miserStat")
	{
		withoutRecord.GET("getCardStat", miserStatApi.GetCardStat)
		withoutRecord.GET("getPieStat", miserStatApi.GetPieStat)
		withoutRecord.GET("getLineStat", miserStatApi.GetLineStat)
		withoutRecord.GET("getMonthStat", miserStatApi.GetMonthStat)
		withoutRecord.GET("getYearStat", miserStatApi.GetYearStat)
		withoutRecord.GET("getItemStat", miserStatApi.GetItemStat)
		withoutRecord.GET("getCategoryStat", miserStatApi.GetCategoryStat)
		withoutRecord.GET("getMonthTransactionStat", miserStatApi.GetMonthTransactionStat)
		withoutRecord.GET("getCategoryItemStat", miserStatApi.GetCategoryItemStat)
	}

	withoutAuth := publicRouter.Group("miserStat")
	{
		_ = withoutAuth
	}
}

File: web/src/api/miser/miser_stat.js
import service from '@/utils/request'

export const getCardStat = (params) => {
  return service({
    url: 'miserStat/getCardStat',
    method: 'get',
    params
  })
}

export const getPieStat = (params) => {
  return service({
    url: 'miserStat/getPieStat',
    method: 'get',
    params
  })
}

export const getLineStat = (params) => {
  return service({
    url: 'miserStat/getLineStat',
    method: 'get',
    params
  })
}

export const getMonthStat = (params) => {
  return service({
    url: 'miserStat/getMonthStat',
    method: 'get',
    params
  })
}

export const getYearStat = (params) => {
  return service({
    url: 'miserStat/getYearStat',
    method: 'get',
    params
  })
}

export const getItemStat = (params) => {
  return service({
    url: 'miserStat/getItemStat',
    method: 'get',
    params
  })
}

export const getCategoryStat = (params) => {
  return service({
    url: 'miserStat/getCategoryStat',
    method: 'get',
    params
  })
}

export const getMonthTransactionStat = (params) => {
  return service({
    url: 'miserStat/getMonthTransactionStat',
    method: 'get',
    params
  })
}

export const getCategoryItemStat = (params) => {
  return service({
    url: 'miserStat/getCategoryItemStat',
    method: 'get',
    params
  })
}

export const getRankingStat = (params) => {
  return service({
    url: 'miserStat/getRankingStat',
    method: 'get',
    params
  })
}
